es74x-cloud/src/main/scala/tech/beshu/ror/proxy/es/clients/HighLevelClientBasedClusterAdminClient.scala

```diff
@@ -0,0 +1,277 @@
+import org.elasticsearch.action.admin.cluster.stats.{ClusterStatsRequest, ClusterStatsRequestBuilder, ClusterStatsResponse}
+import org.elasticsearch.action.admin.cluster.storedscripts.{DeleteStoredScriptRequest, DeleteStoredScriptRequestBuilder, GetStoredScriptRequest, GetStoredScriptRequestBuilder, GetStoredScriptResponse, PutStoredScriptRequest, PutStoredScriptRequestBuilder}
+import org.elasticsearch.action.admin.cluster.tasks.{PendingClusterTasksRequest, PendingClusterTasksRequestBuilder, PendingClusterTasksResponse}
+import org.elasticsearch.action.ingest.{DeletePipelineRequest, DeletePipelineRequestBuilder, GetPipelineRequest, GetPipelineRequestBuilder, GetPipelineResponse, PutPipelineRequest, PutPipelineRequestBuilder, SimulatePipelineRequest, SimulatePipelineRequestBuilder, SimulatePipelineResponse}
+import org.elasticsearch.action.support.master.AcknowledgedResponse
+import org.elasticsearch.client.ClusterAdminClient
+import org.elasticsearch.cluster.{ClusterName, ClusterState}
+import org.elasticsearch.common.bytes.BytesReference
+import org.elasticsearch.common.xcontent.XContentType
+import org.elasticsearch.tasks.TaskId
+import org.elasticsearch.threadpool.ThreadPool
+
+class HighLevelClientBasedClusterAdminClient(esClient: RestHighLevelClientAdapter)
+                                            (implicit scheduler: Scheduler)
+  extends ClusterAdminClient {
+
+  override def health(request: ClusterHealthRequest): ActionFuture[ClusterHealthResponse] = ???
+
+  override def health(request: ClusterHealthRequest, listener: ActionListener[ClusterHealthResponse]): Unit = {
+    esClient
+      .health(request)
+      .runAsync(handleResultUsing(listener))
+  }
+
+  override def prepareHealth(indices: String*): ClusterHealthRequestBuilder = ???
+
+  override def state(request: ClusterStateRequest): ActionFuture[ClusterStateResponse] = ???
+
+  override def state(request: ClusterStateRequest, listener: ActionListener[ClusterStateResponse]): Unit = {
+    // todo: implement properly
+    listener.onResponse(new ClusterStateResponse(
+      ClusterName.DEFAULT,
+      ClusterState.EMPTY_STATE,
+      false
+    ))
+  }
+
+  override def prepareState(): ClusterStateRequestBuilder = ???
+
+  override def updateSettings(request: ClusterUpdateSettingsRequest): ActionFuture[ClusterUpdateSettingsResponse] = ???
+
+  override def updateSettings(request: ClusterUpdateSettingsRequest, listener: ActionListener[ClusterUpdateSettingsResponse]): Unit = ???
+
+  override def prepareUpdateSettings(): ClusterUpdateSettingsRequestBuilder = ???
+
+  override def prepareReloadSecureSettings(): NodesReloadSecureSettingsRequestBuilder = ???
+
+  override def reroute(request: ClusterRerouteRequest): ActionFuture[ClusterRerouteResponse] = ???
+
+  override def reroute(request: ClusterRerouteRequest, listener: ActionListener[ClusterRerouteResponse]): Unit = ???
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Usage of ??? operator](https://app.codacy.com/app/scarduzio/elasticsearch-readonlyrest-plugin/pullRequest?prid=4591272)